### PR TITLE
feat: local sonarqube support

### PR DESCRIPTION
**Pull request by teolemon**
### What
- Add local sonarqube support
- https://github.com/sonar-perl/sonar-perl

### Part of
- #5512

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# must be unique in a given SonarQube instance
+sonar.projectKey=My::Project
+# this is the name displayed in the SonarQube UI
+sonar.projectName=My Project
```
Comment (teolemon) [inline sonar-project.properties:4]:
```suggestion
sonar.projectName=openfoodfacts-server
```

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# If not set, SonarQube starts looking for source code from the directory containing 
+# the sonar-project.properties file.
+sonar.sources=lib
+ 
+# Encoding of the source code. Default is default system encoding
+sonar.sourceEncoding=UTF-8
+
+sonar.tests=t
+# testReport.tgz is the default for the report archive.
+sonar.perl.testHarness.archivePath=testReport.tgz
```
Comment (hangy) [inline sonar-project.properties:18]: Maybe add to `.gitignore`?

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+sonar.coverageReportPaths=cover_db/sonar_generic.xml
```
Comment (hangy) [inline sonar-project.properties:20]: Maybe add to `.gitignore`?

File: sonar-project.properties [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# must be unique in a given SonarQube instance
+sonar.projectKey=My::Project
```
Comment (VaiTon) [inline sonar-project.properties:2]: Is this the key?

**Comment by VaiTon** [comment]
Also, I saw that sonarcloud does not support Perl, are we going to run our own instance of SonarQube?

**Comment by teolemon** [comment]
I've started trying running my own local instance on my laptop, using docker.
https://github.com/sonar-perl/sonar-perl

**Comment by VaiTon** [comment]
@teolemon could you resolve the issues highlighted if you have time?

**Review by teolemon — COMMENTED** [review]
If I remember well, my original intent was to have more feedback on Perl Code quality (the hosted Sonarqube didn't support Perl)

**Comment by stephanegigandet** [comment]
We already have perlcritic and unit/integration tests so the added value of sonar cube is unclear. I'm closing the PR, we can revisit later if the need arises.

**Comment by teolemon** [comment]
@alexgarel @stephanegigandet

**Pull request closed without merging**